src/net/conn_plain.c

```diff
@@ -25,6 +27,24 @@ set_error(int err)
 #endif
 }
 
+static int
+get_error(void)
+{
+#ifdef WIN32
+	return WSAGetLastError();
+#else
+	return errno;
+#endif
+}
+
+/* We cannot define `pg_strerror` here because there is a #define in PG12 that
+ * sets `strerror` to `pg_strerror`. Instead, we handle the missing case for
+ * pre-PG12 on Windows by setting `strerror` to the windows version of the
+ * function and use `strerror` below. */
+#if PG12_LT && WIN32
+#define strerror pgwin32_socket_strerror
```

Comment:
When dealing with macros or function, I prefer defining a full macro with function parameters instead of doing string replacement. This better protects against accidental string replacement, e.g., if a variable has the same name as the function.

```suggestion
#define strerror (errno) pgwin32_socket_strerror(errno)
```